tests/specs/benchmarks/verification.k

```diff
@@ -298,6 +298,19 @@ rule #ecrecData(SIGS, I, DATA_HASH) => #encodeArgs(#bytes32(DATA_HASH), #uint8(#
 
 endmodule
 
+module VERIFICATION-HASKELL [symbolic, kore]
+    imports VERIFICATION-COMMON
+
+    syntax Int ::= "#infGas" [function, functional]
+ // -----------------------------------------------
+    rule #gas(_, _, _) => #infGas
+
+    rule #infGas  <Int _ => false
+    rule #infGas >=Int _ => true
+
+    rule #infGas -Int _ => #infGas
```

Comment:
`#infGas -Int #infGas` is not necessarily `#infGas`.

The inequality rules above are similarly problematic too.

To be safe, we need to ensure that `#infGas` does not appear in the other side.  That was done using K-REFLECTION in the java backend, but I have no good idea how to do in the haskell backend.

Reply:
Concrete initial gas didn't end up working out because the haskell backend could not prove that `NNNNNNNNN <Int Csstore(_, _, _, _)` for example. But maybe we can add lemmas for that.

By the way, does `rule #infGas <Int G => false requires G =/=Int #infGas` not work for some reason?

Reply:
Not including Csstore in the concrete-rules.txt, z3 should be able to prove it, if this issue (https://github.com/kframework/kore/issues/1962) was properly fixed.

Reply:
what If G is foo(#infGas)?

Reply:
Ok, instead I've only simplified it in cases where (i) it's needed and (ii) it's safe.